pkg/cmd/util/clientcmd/factory.go

```diff
@@ -166,6 +169,31 @@ func NewFactory(clientConfig kclientcmd.ClientConfig) *Factory {
 	w.Printer = func(mapping *meta.RESTMapping, noHeaders, withNamespace, wide bool, showAll bool, columnLabels []string) (kubectl.ResourcePrinter, error) {
 		return describe.NewHumanReadablePrinter(noHeaders, withNamespace, wide, showAll, columnLabels), nil
 	}
+	kAttachablePodForObjectFunc := w.Factory.AttachablePodForObject
+	w.AttachablePodForObject = func(object runtime.Object) (*api.Pod, error) {
+		_, kc, err := w.Clients()
+		if err != nil {
+			return nil, err
+		}
+		switch t := object.(type) {
+		case *deployapi.DeploymentConfig:
+			var pods *api.PodList
+			for pods == nil || len(pods.Items) == 0 {
+				var err error
+				if pods, err = kc.Pods(t.Namespace).List(labels.SelectorFromSet(t.Template.ControllerTemplate.Selector), fields.Everything()); err != nil {
+					return nil, err
+				}
+				if len(pods.Items) == 0 {
```

Comment:
Shouldn't we sort here and pick the oldest pod? See https://github.com/openshift/origin/issues/4201#issuecomment-131993792